src/Repository/UserRepository.php

```diff
@@ -136,12 +137,18 @@ public function findAllMatchingDTOsByCampusIds(array $campusIds): array
     public function findEventsForUser($id, DateTime $from, DateTime $to): array
     {
         //These joins are DQL representations to go from a user to an offerings
-        $joins = $this->getUserToOfferingJoins();
+        $joinsAndUserContexts = $this->getUserToOfferingJoinsAndUserContexts();
 
         $offeringEvents = [];
         //using each of the joins above create a query to get events
-        foreach ($joins as $join) {
-            $groupEvents = $this->getOfferingEventsFor($id, $from, $to, $join);
+        foreach ($joinsAndUserContexts as $joinsAndUserContext) {
+            $groupEvents = $this->getOfferingEventsFor(
+                $id,
+                $from,
+                $to,
+                $joinsAndUserContext[0],
+                $joinsAndUserContext[1],
```

Comment:
I'm not loving the array by index here. It's completely self contained in this file, and probably easier to understand that I'm feeling, but reading it a few times it's hard to follow which of these is which. I'm mainly ruminating, I'm ok to keep it.

Reply:
it's gnarly.